common/models/user.js

```diff
@@ -454,6 +454,22 @@ module.exports = function(User) {
   };
 
   /*!
+   * Hash the plain password
+   */
+  User.hashPassword = function(plain) {
+    this.validatePassword(plain);
+    var salt = bcrypt.genSaltSync(this.settings.saltWorkFactor || SALT_WORK_FACTOR);
+    return bcrypt.hashSync(plain, salt);
+  };
+
+  User.validatePassword = function(plain) {
+    if (typeof plain === 'string' && plain) {
+      return true;
+    }
+    throw new Error('Invalid password: ' + plain);
```

Comment:
How `throw new Error` going to integrate with the existing validations? The request setting invalid password should return HTTP status 422 and a description of the validation error in the usual format. This code will IMO return 500 without enough details.